vendor/vendor-android/adam/src/main/kotlin/com/malinskiy/marathon/android/adam/log/LogCatHeader.kt

```diff
@@ -0,0 +1,105 @@
+package com.malinskiy.marathon.android.adam.log
+
+import java.time.Instant
+import java.util.*
+
+/**
+ * Data class for message header information which gets reported by logcat.
+ */
+class LogCatHeader {
+    val logLevel: Log.LogLevel
+    val pid: Int
+    val tid: Int
+    val appName: String
+    val tag: String
+    val timestampInstant: Instant?
+    private val mTimestamp: LogCatTimestamp?
+
+    constructor(
+        logLevel: Log.LogLevel,
+        pid: Int,
+        tid: Int,
+        appName: String,
+        tag: String,
+        timestampInstant: Instant
+    ) {
+        this.logLevel = logLevel
+        this.pid = pid
+        this.tid = tid
+        this.appName = appName
+        this.tag = tag
+        this.timestampInstant = timestampInstant
+        mTimestamp = null
+    }
+
+    /**
+     * Construct an immutable log message object.
+     *
+     */
+    @Deprecated("Use {@link #LogCatHeader(LogLevel, int, int, String, String, Instant)}")
```

Comment:
Can you remove deprecated constructor and make it data class?

Reply:
The deprecated constructor is actually used when parsing logs. I'll work on cleaning it up, but not sure if I'll be able to use the other constructor

Reply:
Reworked